code/go/0chain.net/core/util/mpt_nodedb_test.go

```diff
@@ -1002,3 +1004,335 @@ func TestRaceLevelNodeDB_Full(t *testing.T) {
 		require.Len(t, lndb.DeletedNodes, 0)
 	})
 }
+
```

Comment:
There are closure issues in the existing code. (Issue https://github.com/0chain/0chain/issues/266)
[parrellel access](https://github.com/0chain/0chain/blob/master/code/go/0chain.net/core/util/mpt_nodedb_test.go#L463)
```go
for i := 0; i < parallel; i++ {
    t.Run("parallel access", func(t *testing.T) {
        t.Parallel()

        lndb.RebaseCurrentDB(fcurr)

        if lndb.GetPrev() == prev {
            lndb.SetPrev(fprev)
        } else {
            lndb.SetPrev(prev)
        }

        for j := 0; j < len(kvs); j++ {
            switch (i + j) % 3 {
```
Needs `i<=i`. When I tried this the test failed, so might need some extra work to get the test to pass. 

[Also](https://github.com/0chain/0chain/blob/master/code/go/0chain.net/core/util/mpt_nodedb_test.go#L723) 
```go
for _, ndb := range []NodeDB{
    pndb,
    lndb,
    mndb,
} {
    t.Run(fmt.Sprintf("%T", ndb), func(t *testing.T) {
        for i := 0; i < parallel; i++ {
            t.Run("parallel access", func(t *testing.T) {
                t.Parallel()
                for j := 0; j < len(kvs); j++ {
                    var err error
                    switch (i + j) % 6 {
                    case 0:
                        _, err = ndb.GetNode(kvs[j].key)
```
So needs `ndb := ndb` and `i:=i`.

Reply:
Fix skip or delete, but there is no point in having these tests run as they are.